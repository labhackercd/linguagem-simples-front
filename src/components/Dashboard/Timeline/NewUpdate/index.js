import React, {useState, useEffect} from 'react'
import {Paper,Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText,
	      DialogTitle, Grid, Typography, Box,  List, ListItem} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
import DividerIcon from './../../../../assets/divider.svg';
import PictureUploadIcon from './../../../../assets/picture_upload.svg';
import TwitterIcon from './../../../../assets/twitter_icon.svg';
import LinkIcon from './../../../../assets/link.svg'
import { parseHourMinute } from './../../../Util';

const useStyles = makeStyles((theme) => ({
	summaryBox: {
		display: 'flex',
		flexDirection: 'row',
		width: '100%',
	},
	summaryHeader: {
  	display: 'flex',
		margin: '1rem 0 1rem 0',
  },
  textArea: {
    height: '100%',
    margin: '0rem 0 0 0',
    width: '100%',
    border: '0',
  },
  textField: {
		margin: '0 1rem 0 0',
		height: '100%',
		width: '100%',
	},
  submenu: {
    display: 'flex',
  },
  subMenuItem: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    margin: '0 0.5rem 0 1rem',
  },
  button: {
    height: '70%',
    alignSelf: 'flex-end',
  },
  notchedOutline: {
    border: '2px solid #F2F2F2',
    borderWidthBottom: '0px',
    color: theme.palette.secondary,
    borderRadius: '0 0 0 0',
  },
  time: {
    color: theme.palette.primary.main
  },
}))
export default function NewUpdate(props) {
  const classes = useStyles();
	const dateAux = props.date ? parseHourMinute(props.date) : parseHourMinute(new Date())
	const [date, setDate] = useState(dateAux)

	useEffect(() => {
		setInterval(() => tick(), 1000)
	}, clearInterval(1000));

	const tick = () => {
		setDate(parseHourMinute(new Date()))
	}

  return (
    <Grid container className={classes.summaryBox}>
      <Grid container className={classes.summaryHeader}>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-start'}}>
          <Typography variant="h5"> Nova atualização </Typography>
        </Grid>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-end'}}>
        </Grid>
      </Grid>
      <Grid item md={12}>
          <Grid container >
              <form className={classes.textArea} noValidate autoComplete="off">
                <TextField
                  id="newUpdateTextField"
                  multiline
                  rows={4}
									disabled={!props.broadcastingOnline}
                  variant="outlined"
                  className={classes.textField}
                  bgcolor="white"
                  name = "updateText"
                  value={props.updateTextArea}
                  onChange = {props.handleChange}
                  InputProps={{
                    classes: {
                      notchedOutline: classes.notchedOutline
                    },
                  }}
                />
              </form>
          </Grid>
          <Box borderTop={1} color="#F2F2F2" borderRadius="0 0 10px 25px" bgcolor="#F2F2F2">
            <Grid container>
              <Grid item xs={8} className={classes.submenu}>
                <div className={classes.subMenuItem}>
                  <Typography className={classes.time} variant="h6"> {date} </Typography>
                </div>
                <div className={classes.subMenuItem}>
                  <img src={DividerIcon} alt="divider icon"/>
                </div>
                <div className={classes.subMenuItem}>
                  <a href="/"><img src={PictureUploadIcon} id={"picture-upload-icon"} alt="upload icon" onClick={(e) => props.handleDialogStateAction(e, true, "previewDialog", "InputImage")}/></a>
                </div>
								<div className={classes.subMenuItem}>
									<a href="/"><img src={TwitterIcon} id={"tweet-insert-icon"}alt="incorporate twitter icon" onClick={(e) => props.handleDialogStateAction(e, true, "URLInputDialog", "URLInputIsTwitter")}/></a>
								</div>
								<div className={classes.subMenuItem}>
									<a href="/"><img src={LinkIcon} id={"link-insert-icon"} alt="incorporate url icon" onClick={(e) => props.handleDialogStateAction(e, true, "URLInputDialog", "noAction")}/></a>
								</div>
              </Grid>
              <Grid item xs={4} style={{display: 'flex', justifyContent: 'flex-end'}}>
                <Button disabled={!props.broadcastingOnline} id={"updateSubmitButton"}className={classes.button} onClick={props.handleClick} variant="contained" disableElevation>
                  Inserir
                </Button>
              </Grid>
            </Grid>
          </Box>
      </Grid>
    </Grid>
  )
}
