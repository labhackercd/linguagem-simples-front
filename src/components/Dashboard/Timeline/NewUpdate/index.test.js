import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NewUpdate from './index';
import {shallow, mount } from "enzyme/build";
import ReactDOM from 'react-dom';
import MockTheme from './../mockTheme';

it("snapshot should not have differences", () => {
    const component = shallow(<MockTheme><NewUpdate date={new Date(98, 1)}/></MockTheme>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

test('Test if NewUpdate renders without crash', () => {
    const div = document.createElement("div")
    ReactDOM.render(<MockTheme><NewUpdate></NewUpdate></MockTheme>, div)
    ReactDOM.unmountComponentAtNode(div)
});

test('writes update and submit doesnt break page', () => {
    const wrapper = mount(<MockTheme>
                          <NewUpdate handleDialogStateAction={jest.fn()}/>
                          </MockTheme>);
    const button = wrapper.find("#updateSubmitButton").last();
    const textField = wrapper.find("#newUpdateTextField").last();
    textField.instance().value = "test content"
    textField.simulate("change");
    button.simulate('click');
    wrapper.unmount();
});

test('clicking content insertion icons doesnt break page', () => {
  const wrapper = mount(<MockTheme><NewUpdate handleDialogStateAction={jest.fn()}/></MockTheme>);
  const imageInsert = wrapper.find('#picture-upload-icon').last()
  const twitterInsert = wrapper.find('#tweet-insert-icon').last()
  const linkInsert = wrapper.find('#link-insert-icon').last()
  imageInsert.simulate('click')
  twitterInsert.simulate('click')
  linkInsert.simulate('click')
})
