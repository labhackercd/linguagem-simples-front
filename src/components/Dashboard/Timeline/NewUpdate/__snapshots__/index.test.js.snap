// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should not have differences 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-summaryBox-1"
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-summaryHeader-2"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-start",
        }
      }
    >
      <WithStyles(ForwardRef(Typography))
        variant="h5"
      >
         Nova atualização 
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-end",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={12}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <form
        autoComplete="off"
        className="makeStyles-textArea-3"
        noValidate={true}
      >
        <WithStyles(ForwardRef(TextField))
          InputProps={
            Object {
              "classes": Object {
                "notchedOutline": "makeStyles-notchedOutline-8",
              },
            }
          }
          bgcolor="white"
          className="makeStyles-textField-4"
          id="textfield"
          multiline={true}
          name="updateText"
          rows={4}
          variant="outlined"
        />
      </form>
    </WithStyles(ForwardRef(Grid))>
    <Styled(MuiBox)
      bgcolor="#F2F2F2"
      borderRadius="0 0 10px 25px"
      borderTop={1}
      color="#F2F2F2"
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          className="makeStyles-submenu-5"
          item={true}
          xs={8}
        >
          <div
            className="makeStyles-subMenuItem-6"
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-time-9"
              variant="h6"
            >
               18:00 
            </WithStyles(ForwardRef(Typography))>
          </div>
          <div
            className="makeStyles-subMenuItem-6"
          >
            <img
              alt="divider icon"
              src="../../img/divider.svg"
            />
          </div>
          <div
            className="makeStyles-subMenuItem-6"
          >
            <a
              href="/"
            >
              <img
                alt="upload icon"
                src="../../img/picture_upload.svg"
              />
            </a>
          </div>
          <div
            className="makeStyles-subMenuItem-6"
          >
            <img
              alt="incorporate tweet icon"
              src="../../img/twitter_icon.svg"
            />
          </div>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
          xs={4}
        >
          <WithStyles(ForwardRef(Button))
            className="makeStyles-button-7"
            disableElevation={true}
            variant="contained"
          >
            Inserir
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
